lib/bundler/compact_index_client/updater.rb

```diff
@@ -28,55 +28,59 @@ def initialize(fetcher)
       def update(local_path, remote_path, retrying = nil)
         headers = {}
 
-        Dir.mktmpdir("bundler-compact-index-") do |local_temp_dir|
-          local_temp_path = Pathname.new(local_temp_dir).join(local_path.basename)
-
-          # first try to fetch any new bytes on the existing file
-          if retrying.nil? && local_path.file?
-            FileUtils.cp local_path, local_temp_path
-            headers["If-None-Match"] = etag_for(local_temp_path)
-            headers["Range"] =
-              if local_temp_path.size.nonzero?
-                # Subtract a byte to ensure the range won't be empty.
-                # Avoids 416 (Range Not Satisfiable) responses.
-                "bytes=#{local_temp_path.size - 1}-"
-              else
-                "bytes=#{local_temp_path.size}-"
-              end
-          else
-            # Fastly ignores Range when Accept-Encoding: gzip is set
-            headers["Accept-Encoding"] = "gzip"
-          end
+        Bundler::SharedHelpers.filesystem_access(Dir.tmpdir, :write) do
```

Comment:
In that case I can create the temp dir with ```Dir.mktmpdir("bundler-compact-index-")``` before calling ```filesystem_access```, and then pass it to ```filesystem_access```.